### PR TITLE
Add support for Hibernate `6.6.0.Final`

### PR DESCRIPTION
**Pull request by sdeleuze**
Please wait before merging it we have understood another issue we see with Spring. Will post an update later.

### CODE REVIEW

**Comment by sdeleuze** [comment]
Some entries need to be excluded before this PR is merged, will push an update shortly.

**Comment by dnestoro** [comment]
> Some entries need to be excluded before this PR is merged, will push an update shortly.

Sure, I will wait for your update, but before we merge this PR, can we remove gradle stuff added to this library testing (all other tests are using top level gradle infrastructure)?

File: tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/gradle/wrapper/gradle-wrapper.jar
Comment (olpaw) [reply tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/gradle/wrapper/gradle-wrapper.jar:1]: Looks like leftovers that were not removed before creating the PR.

Comment (sdeleuze) [reply tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/gradle/wrapper/gradle-wrapper.jar:1]: Those files are needed to run the `generate-metadata.sh` script, but I can use instead the root ones so I will modify the PR accordingly.

**Comment by sdeleuze** [comment]
@dnestoro Done, I introduced it as it was needed by `generate-metadata.sh` but was able to refine the script to use the root one.

As the script seems to generate far more metadata, it chose to just add missing entries for `[Lorg.hibernate.event.spi.PostUpsertEventListener;` and `org.hibernate.internal.log.ConnectionInfoLogger_$logger` in order to limit the risk of regressions.

Please wait @christophstrobl green light before merging it.

**Comment by christophstrobl** [comment]
Should we add `"default-for": "6\\.5\\..*"` to the `metadata-version` targeting hibernate `6.5.0.Final` line?

this one also showed up in the diff comparing metadata to 6.5:
```json
{
  "condition" : {
    "typeReachable" : "org.hibernate.dialect.function.array.AbstractArrayContainsFunction"
  },
  "name" : "org.hibernate.internal.log.DeprecationLogger_$logger",
  "methods" : [ {
    "name" : "<init>",
    "parameterTypes" : [ "org.jboss.logging.Logger" ]
  } ]
}
```

might be covered by which is in place.
```json
{
  "condition":{"typeReachable":"org.hibernate.internal.log.DeprecationLogger"},
  "name":"org.hibernate.internal.log.DeprecationLogger_$logger",
  "methods":[{"name":"<init>","parameterTypes":["org.jboss.logging.Logger"] }]
},
```

File: tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/src/test/java/org_hibernate_orm/hibernate_core/AbstractHibernateTest.java
Comment (dnestoro) [inline tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/src/test/java/org_hibernate_orm/hibernate_core/AbstractHibernateTest.java:55]: Minor: Don't we need null check here (same as you have done [here](https://github.com/oracle/graalvm-reachability-metadata/pull/523/files#diff-7cd6280680c74063a7f29799f5b1a2c0069e8d6180322d53fd3fce03872b8c60R31))

**Review by dnestoro — APPROVED** [review]

**Comment by dnestoro** [comment]
LGTM, but since the PR is huge, please wait for a second review as well. cc @olpaw

**Review by olpaw — APPROVED** [review]

**Comment by olpaw** [comment]
I looked at the dirdiff for `6.5.0.Final` vs. `6.6.0.Final` in `metadata` and `tests`. The changes are minimal/trivial. No issues.

File: tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/src/test/java/org_hibernate_orm/hibernate_core/Target_BytecodeProvider.java
Comment (olpaw) [inline tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/src/test/java/org_hibernate_orm/hibernate_core/Target_BytecodeProvider.java:17]: @christophstrobl I see that you introduced those substitutions for testing `org.hibernate.orm` `hibernate.core`.
Why exactly are you using them and are you sure that they do not cause the test to be deviating from real-world NI use of  `org.hibernate.orm` `hibernate.core`?

Comment (christophstrobl) [reply tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/src/test/java/org_hibernate_orm/hibernate_core/Target_BytecodeProvider.java:17]: Hibernate had removed the `hibernate.bytecode.provider` option that allowed to set it to `none`.

File: tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/src/test/java/org_hibernate_orm/hibernate_core/AbstractHibernateTest.java
Comment (sdeleuze) [reply tests/src/org.hibernate.orm/hibernate-core/6.6.0.Final/src/test/java/org_hibernate_orm/hibernate_core/AbstractHibernateTest.java:55]: I have just copied Hibernate 6.5 tests.

**Pull request merged**
